### PR TITLE
Implemented the Navbar design 

### PR DESCRIPTION
**Pull request by Akashsah2003**
pr-rs-11
pr-rs-12
pr-rs-13
pr-rs-14

Related issue : #2 
Refactored Navbar

### CODE REVIEW

**Comment by Anurag607** [comment]
the dark mode icon is too big make it smaller.
also keep it's background transparent.
add some gap between the dark mode icon and sign up button

**Comment by Anurag607** [comment]
use monument extended font for "UniShare" (just add the classname "monu" to it) 
and add some horizontal padding to the navbar (use px-10) as well as increase the vertical padding to it as well

File: tailwind.config.js
Comment (Anurag607) [inline tailwind.config.js:1]: change darkMode: 'class', to darkMode: ['class', '[data-theme="dark"]'],

File: src/components/NavBar/index.jsx
Comment (Anurag607) [inline src/components/NavBar/index.jsx:1]: you can comment dark mode functionality (useEffect()) someone is already working on it on another issue

File: src/components/NavBar/index.jsx [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+  }, [darkMode]);
+
+  const toggleDarkMode = () => {
+    setDarkMode(!darkMode);
+  };
+
+
   return (
-    <div>NavBar</div>
+    <div className='sticky'>
```
Comment (Anurag607) [inline src/components/NavBar/index.jsx:25]: use flex here to center the child div inside it and keep it's width as w-screen

File: src/components/NavBar/index.jsx [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+      <div className='flex rounded-xl shadow-md mx-3 my-4 bg-gray-100 dark:bg-gray-300 dark:text-white'>
```
Comment (Anurag607) [inline src/components/NavBar/index.jsx:26]: keep it's width 90vw

File: src/components/NavBar/index.jsx [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+        </div>
+        <div className='flex w-3/4 flex-row-reverse items-center'>
+          <button className='h-10 bg-white mx-2 rounded-lg shadow-md dark:bg-gray-400' onClick={toggleDarkMode}>
+            <svg className='h-full' viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><g id="SVGRepo_bgCarrier" stroke-width="0"></g><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g><g id="SVGRepo_iconCarrier"> <path className='fill-blue-500 dark:fill-yellow-500' d="M12.0557 3.59974C12.2752 3.2813 12.2913 2.86484 12.0972 2.53033C11.9031 2.19582 11.5335 2.00324 11.1481 2.03579C6.02351 2.46868 2 6.76392 2 12C2 17.5228 6.47715 22 12 22C17.236 22 21.5313 17.9764 21.9642 12.8518C21.9967 12.4664 21.8041 12.0968 21.4696 11.9027C21.1351 11.7086 20.7187 11.7248 20.4002 11.9443C19.4341 12.6102 18.2641 13 17 13C13.6863 13 11 10.3137 11 6.99996C11 5.73589 11.3898 4.56587 12.0557 3.59974Z" fill="#000000"></path> </g></svg>
+          </button>
+          <button className='rounded-lg bg-[radial-gradient(ellipse_at_center,_var(--tw-gradient-stops))] from-blue-500 to-blue-800 text-white my-2 px-2 py-1 font-bold shadow-md'>Sign Up</button>
```
Comment (Anurag607) [inline src/components/NavBar/index.jsx:34]: remove the radial gradient...if you want you can get the css for box shadow from the figma design

**Review by Anurag607 — CHANGES_REQUESTED** [review]

File: src/components/NavBar/index.jsx (hunk 3 above)
Comment (Anurag607) [reply src/components/NavBar/index.jsx:34]: you can it via the notion link mentioned in the readme

Comment (Akashsah2003) [reply src/components/NavBar/index.jsx]: Done

File: src/components/NavBar/index.jsx (hunk 2 above)
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx]: Done

File: src/components/NavBar/index.jsx (hunk 1 above)
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx]: Done

File: src/components/NavBar/index.jsx
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx:1]: Done

File: tailwind.config.js
Comment (Akashsah2003) [reply tailwind.config.js:1]: Done

**Comment by Anurag607** [comment]
image
could we have the a popup like this for native devices 🤔? This dropdown makes the screen look a bit crowded

File: src/components/NavBar/index.jsx [hunk 4]
```diff
@@ -1,8 +1,67 @@
...
+
   return (
-    <div>NavBar</div>
+    <div className='sticky flex w-screen justify-center flex-col items-center'>
+      <div className='flex rounded-2xl shadow-[0px_6px_8px_rgba(0,0,0,0.25)] w-[90vw] mx-3 my-4 px-10 py-1 bg-[#F4F4F5] dark:bg-[#44403C] dark:text-white'>
+        <div className='flex w-1/4 monu font-normal px-2 py-2 text-xl items-center'>
+          <Link to='/'>UniShare</Link>
+        </div>
+        <div className='mobile:hidden flex w-3/4 flex-row-reverse items-center'>
+          <button className='h-7 bg-transparent mx-2 rounded-lg shadow-md' onClick={toggleDarkMode}>
```
Comment (Anurag607) [inline src/components/NavBar/index.jsx:36]: define height and width for the dark mode in terms of px: h-[32px] w-[32px] and use "grid place-items-center" to center the svg inside it

**Comment by Anurag607** [comment]
there are certain errors like this, please resolve them as well:
image

**Comment by Akashsah2003** [comment]
pr-rs-131
pr-rs-132
pr-rs-133
pr-rs-134

Added animated hamburger button. Adjusted menu to the right closer to button. Centred dark icon. Resolved console errors.

File: src/components/NavBar/index.jsx (hunk 4 above)
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx:35]: Added height width and used flex box to center it.

**Comment by Akashsah2003** [comment]
@Anurag607 I did the changes as asked. Please review the changes.

**Review by Anurag607 — APPROVED** [review]

**Comment by Anurag607** [comment]
I think it looks good....
@sambhavgupta0705 PTAL, if everything is alright, will merge thne

**Comment by Akashsah2003** [comment]
pr-rs-141
pr-rs-142
pr-rs-143
pr-rs-144
pr-rs-145
pr-rs-146

Latest navbar after the commit

File: src/components/NavBar/index.jsx [hunk 5]
```diff
@@ -1,8 +1,67 @@
...
+
+  const open = () => {
+    setOpenMenu(!openmenu);
+  }
+
+
   return (
-    <div>NavBar</div>
+    <div className='sticky flex w-screen justify-center flex-col items-center'>
+      <div className='flex rounded-2xl shadow-[0px_6px_8px_rgba(0,0,0,0.25)] w-[90vw] mx-3 my-4 px-10 py-1 bg-[#F4F4F5] dark:bg-[#44403C] dark:text-white'>
```
[the remark this answers is not in the document]
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx]: Done

File: src/components/NavBar/index.jsx [hunk 6]
```diff
@@ -1,8 +1,67 @@
...
+        <div className='flex w-1/4 monu font-normal px-2 py-2 text-xl items-center'>
+          <Link to='/'>UniShare</Link>
+        </div>
+        <div className='mobile:hidden flex w-3/4 flex-row-reverse items-center'>
+          <button className='h-7 bg-transparent mx-2 rounded-lg shadow-md' onClick={toggleDarkMode}>
+            <svg className='h-full' viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><g id="SVGRepo_bgCarrier" stroke-width="0"></g><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g><g id="SVGRepo_iconCarrier"> <path className='fill-[#3B82F6] dark:fill-yellow-500' d="M12.0557 3.59974C12.2752 3.2813 12.2913 2.86484 12.0972 2.53033C11.9031 2.19582 11.5335 2.00324 11.1481 2.03579C6.02351 2.46868 2 6.76392 2 12C2 17.5228 6.47715 22 12 22C17.236 22 21.5313 17.9764 21.9642 12.8518C21.9967 12.4664 21.8041 12.0968 21.4696 11.9027C21.1351 11.7086 20.7187 11.7248 20.4002 11.9443C19.4341 12.6102 18.2641 13 17 13C13.6863 13 11 10.3137 11 6.99996C11 5.73589 11.3898 4.56587 12.0557 3.59974Z" fill="#000000"></path> </g></svg>
```
[the remark this answers is not in the document]
Comment (Akashsah2003) [reply src/components/NavBar/index.jsx]: Done

**Review by sambhavgupta0705 — APPROVED** [review]
@Anurag607 LGTM,
@Akashsah2003 Great Work and thanks for coping with us with patience.
Hope to see your contribution for other issues also.

**Pull request merged**